cpython/patches/0001-call-trampolines-to-handle-fpcast-troubles.patch

```diff
@@ -0,0 +1,179 @@
+From a608eb640f151f08ffd810e53eb29f58ef68f493 Mon Sep 17 00:00:00 2001
+From: Hood Chatham <roberthoodchatham@gmail.com>
+Date: Fri, 3 Dec 2021 20:55:40 -0800
+Subject: [PATCH] call trampolines to handle fpcast troubles
+
+---
+ Objects/call.c         |  5 ++++-
+ Objects/descrobject.c  | 28 +++++++++++++++++++---------
+ Objects/methodobject.c | 12 +++++++++---
+ 3 files changed, 32 insertions(+), 13 deletions(-)
+
+diff --git a/Objects/call.c b/Objects/call.c
+index 87dc0dbbdb..f8c7811521 100644
+--- a/Objects/call.c
++++ b/Objects/call.c
+@@ -143,6 +143,9 @@ PyObject_VectorcallDict(PyObject *callable, PyObject *const *args,
+ }
+ 
+ 
++// Defined in methodobject patch
++PyObject* method_call_trampoline(PyCFunction func, PyObject *self, PyObject *arg1, PyObject *arg2);
++
+ PyObject *
+ _PyObject_MakeTpCall(PyThreadState *tstate, PyObject *callable,
+                      PyObject *const *args, Py_ssize_t nargs,
+@@ -188,7 +191,7 @@ _PyObject_MakeTpCall(PyThreadState *tstate, PyObject *callable,
+     PyObject *result = NULL;
+     if (_Py_EnterRecursiveCall(tstate, " while calling a Python object") == 0)
+     {
+-        result = call(callable, argstuple, kwdict);
++        result = method_call_trampoline((PyCFunction)call, callable, argstuple, kwdict);
+         _Py_LeaveRecursiveCall(tstate);
+     }
+ 
+diff --git a/Objects/descrobject.c b/Objects/descrobject.c
+index fce9cdd309..326844f569 100644
+--- a/Objects/descrobject.c
++++ b/Objects/descrobject.c
+@@ -174,6 +174,11 @@ member_get(PyMemberDescrObject *descr, PyObject *obj, PyObject *type)
+     return PyMember_GetOne((char *)obj, descr->d_member);
+ }
+ 
++#include <emscripten.h>
++EM_JS(PyObject*, getter_call_trampoline, (getter get, PyObject *obj, void *closure), {
```

Comment:
This could use more explanations either in the body of the patch or in the patch description